lib/spack/spack/cmd/buildcache.py

```diff
@@ -842,12 +849,21 @@ def migrate_fn(args):
 
 
 def prune_fn(args):
-    """prune stale buildcache entries from the mirror"""
+    """prune buildcache entries from the mirror
+
+    If a keeplist file is provided, performs direct pruning (deletes packages not in keeplist)
+    followed by orphan pruning. If no keeplist is provided, only performs orphan pruning.
+    """
     mirror: spack.mirrors.mirror.Mirror = args.mirror
     dry_run: bool = args.dry_run
     assert isinstance(mirror, spack.mirrors.mirror.Mirror)
 
-    prune(mirror, dry_run)
+    started_at = get_buildcache_normalized_time(mirror)
```

Comment:
Thinking more on the right way to get the pruning boundary time. We should try to use the `mtime` of the cache index manifest, then the the normalized time if that fails. This way we at least try to bound the pruning based on mirror state with the assumption that the index is updated as part of the builder workflow (like how we do it is CI).